spec/requests/medications_spec.rb

```diff
@@ -0,0 +1,293 @@
+
+    def post_create(medication_params)
+      post medications_path, params: { medication: medication_params }
+    end
+
+    context 'when signed in' do
+      before { sign_in user }
+
+      context 'when valid params are supplied' do
+
+        it 'creates a medication' do
+          expect { post_create valid_medication_params }
+            .to change(Medication, :count).by(1)
+        end
+
+        it 'redirects to medication' do
+          post_create valid_medication_params
+          expect(response).to have_http_status(302)
+        end
+      end
+
+      context 'when invalid params are supplied' do
+        let(:invalid_medication_params) do
+          valid_medication_params.merge(name: nil, dosage: nil)
+        end
+
+        context 'when the request is for html' do
+          it 'returns a successful response' do
+            post_create invalid_medication_params
+            expect(response).to be_successful
+          end
+        end
+
+        context 'when the request is for json' do
+          it 'returns 422, unprocessable entity' do
+            post medications_path,
+              params: { medication: invalid_medication_params },
+              headers: { "ACCEPT" => "application/json" }
+            expect(response).to have_http_status(422)
+          end
+        end
+      end
+
+      context 'when the user_id is hacked' do
+        it 'creates a new medication, ignoring the user_id parameter' do
+          # passing a user_id isn't an error, but it shouldn't
+          # affect the owner of the created item
+          another_user = create(:user2)
+          hacked_medication_params =
+            valid_medication_params.merge(user_id: another_user.id)
+          expect { post_create hacked_medication_params }
+            .to change(Medication, :count).by(1)
+          expect(Medication.last.user_id).to eq(user.id)
+        end
+      end
+    end
+
+    context 'when not signed in' do
+      before { post_create valid_medication_params }
+      it_behaves_like :with_no_logged_in_user
+    end
+  end
+
+  describe '#update' do
+    let(:user) { create(:user1) }
+    let!(:medication) { create(:medication, user_id: user.id) }
+    let!(:valid_medication_params) { { name: 'Updated Medication Name' } }
+
+    def put_update(medication_params)
+      put medication_path(medication), params: { medication: medication_params }
+    end
+
+    context 'when signed in' do
+      before { sign_in user }
+
+      context 'when valid params are supplied' do
+
+        it 'updates a medication' do
+          updated_name_slug = valid_medication_params[:name]
+            .downcase
+            .gsub(' ', '-')
+          put_update valid_medication_params
+          expect(response.body).to include(updated_name_slug)
+        end
+
+        it 'redirects to medication' do
+          put_update valid_medication_params
+          expect(response).to have_http_status(302)
```

Comment:
I think `redirect_to` seems to be the better case if the test reads `redirects to medication`, then I'd probably expect it to be the medication page.